enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,378 @@
+>scenarios?  We've also received suggestions around tmpfs or read only bind-mounts of a directory from the CSI driver.
+>**This question most likely can be deferred until implementation.**
+> 3. SELinux implications will need to be properly understood.  Minimally, a consuming Pod needs the correct SELinux 
+>context and fs permissions with bind-mounts for example.
+>**This question most likely can be deferred until implementation.**
+> 4. The EP document in https://github.com/openshift/enhancements/pull/169 needs to be reconciled with this document in
+>some form or fashion.
+
+## Summary
+
+This proposal will describe a [CSI driver](https://github.com/container-storage-interface/spec/blob/master/spec.md) based
+means for OpenShift cluster administrators to share `Secrets` / `ConfigMaps` defined in one namespace with
+other namesapces. 
+
+Any owners/authors/users of Pods in those namespaces can opt into the consumption of the content of those shared `Secrets` / 
+ `ConfigMaps` during their Pod's execution.
+ 
+The precise means of opting into consuming the shared content are outside the scope of this proposal.
+
+This proposal was broken off from [a proposal on injecting RHEL entitlements/subscriptions](https://github.com/openshift/enhancements/pull/214)
+**EDITORIAL NOTE:** if/when PR 214 merges the reference link will be adjusted to point to the merged document hosted on
+https://github.com/openshift/enhancements.  Or if that PR is replaced by another, this proposal will point to that.
+
+
+## Motivation
+
+Help simplify consumption of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces by providing
+the foundational pieces for building a shared secret/configmap feature such that user pods in one namespace can consume
+secrets/configmaps from another namespace (which is presumably maintained by a cluster admin).
+
+### Goals
+
+Provide easy to use sharing of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces.
+
+### Non-Goals
+
+1. This proposal assumes the sensitive content encapsulated in the `Secret` or `ConfigMap` is valid and correct for the user's purposes.
+No validation of the content for specific usage within a Pod will be done.  That said, this proposal does provide 
+for watching for updates to those objects.
+So if the cluster administrator realized on their own they have provided bad data, and then corrects it, this proposal
+will watch for, discover, and propagate the update when new Pods consuming the content are provisioned.
+
+2. On protecting this content from being viewed/copied by users: if it's available for the Pod to use, the user who can 
+create a Pod that can utilize this feature can also read / view / exfiltrate the Secret contents. It is technically 
+impossible to avoid that.
+
+3. Providing security constraints on who can create these volumes and/or use them in their pods to access secrets/configmaps
```

Comment:
Before such a feature can be considered complete, this needs to be addressed.  This should be a goal. You don't need to solve it for prototypes, but we cannot release software without some kind of restrictions on how content is exposed.  This type of problem dates back to the days of cross-namespace RBAC rolebinding refs. It was a big mistake to allow it.

Reply:
absolutely, see the "GA requrements" at the end of this EP.  We will not ship this CSI driver w/o the additional admission logic that makes it safe to install.

Reply:
(but to be clear, that does not make it a goal of this EP to address that RBAC requirement.  The parent EP i referenced has that goal+responsibility).

Reply:
will reference GA reqs below and clarify this is handled in separate EP

Reply:
> (but to be clear, that does not make it a goal of this EP to address that RBAC requirement. The parent EP i referenced has that goal+responsibility).

I maintain that a prototype really needs to build this in from the start to be viable.  We have experienced many times in this project and the upstream project extreme difficulty in retconing security.  You can see below the impact this has in API design.  It's particularly difficult to consider now to shape the API and should be fairly easy to add to a prototype as a distinct unit.